test/command/5876-datadir-metadata/metadata/test-metadata.yaml

```diff
@@ -0,0 +1,4 @@
+---
```

Comment:
You'll need to make sure this test file is included in extra-source-files in pandoc.cabal, or it won't be included in the source distribution.
Maybe also worth testing two further cases:

- the case where the file exists in the working directory AND in DATADIR/metadata
- the case where the file exists in neither place (yielding an error)

Reply:
Added resources to `extra-source-files` as well as two new tests. There's also the case where the file exists in the working directory, but not in the user directory, which is covered by multiple existing tests. I could still add it under this issue for completeness, though.

Reply:
Note that you shouldn't put 7851.md and the like in extra-source-files, because it's already covered by a glob (`test/command/*.md`).  However, the yaml file in the subdirectory wasn't covered by a glob.  Note that `make check-cabal` will automatically check to make sure that there aren't committed test files that aren't in the extra-source-files stanza.  I don't think we necessarily need the extra test case you describe, but it wouldn't hurt.  Up to you.

Reply:
Great, I ran the check and everything should be included now. I also added a fourth test for when there's no user directory specified and the file does not exist. Then all four possible outcomes of `readMetadataFile` are covered in `5876.md`.